comp/logs/agent/config/endpoints_test.go

```diff
@@ -488,6 +488,37 @@ func (suite *EndpointsTestSuite) TestIsReliableDefaultTrue() {
 	suite.Len(endpoints.GetReliableEndpoints(), 3)
 }
 
+func (suite *EndpointsTestSuite) TestUseSSLDefaultTrue() {
+	var (
+		endpoints *Endpoints
+		err       error
+	)
+
+	suite.config.SetWithoutSource("logs_config.additional_endpoints", []map[string]interface{}{
+		{
+			"host":    "a",
```

Comment:
can we also add a case (or another test) that handles:
`logs_config.logs_no_ssl: true`
`logs_config.logs_no_ssl: false`
`logs_config.logs_no_ssl:` unset

with each permutation of `use_ssl` at the endpoint level? This way we can verify that the setting is correctly overridden (set to the same as the main endpoint) when unset, and overridden by the more specific endpoints settings when needed.